tests/graphicsItems/test_PlotDataItem.py

```diff
@@ -81,7 +81,19 @@ def test_setData():
     pdi.setData([],[])
     assert pdi.xData is None
     assert pdi.yData is None
-    
+
+    # recarray (issue #3275)
+    data = np.recarray((10,), dtype=[('x', float), ('y', float)])
+    pdi.setData(data)
+    assert all(pdi.xData == data["x"])
+    assert all(pdi.yData == data["y"])
```

Comment:
I don't think we should be performing this comparison as recarray is created empty. NANs may have been created. Then the CI may randomly fail? 

Better to create an initialised recarray.

But why even add support for such an input type if it has never worked before? The user could just pass in data.x and data.y as inputs instead?

Reply:
Yep, good point on the uninitialized data - was mostly just using the issue as a test case but there's not much reason to do so.

It's a documented feature, but that is an interesting point that if it hasn't actually worked in the past, we could just remove it. I could go either way on that to be honest.